velox/functions/sparksql/String.h

```diff
@@ -845,4 +845,85 @@ struct TranslateFunction {
   }
 };
 
+/// Decodes `string` in 'application/x-www-form-urlencoded' format using UTF-8
+/// encoding scheme.
+template <typename T>
+struct UrlDecoderFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input does NOT always produce ASCII result.
+  static constexpr bool is_default_ascii_behavior = false;
+
+  int decode(StringView input) {
+    auto toIntFromHex = [](char x) {
+      if (std::isdigit(x)) {
+        return x - '0';
+      }
+      switch (x) {
+        case 'a':
+        case 'A':
+          return 10;
+        case 'b':
+        case 'B':
+          return 11;
+        case 'c':
+        case 'C':
+          return 12;
+        case 'd':
+        case 'D':
+          return 13;
+        case 'e':
+        case 'E':
+          return 14;
+        case 'f':
+        case 'F':
+          return 15;
+        default:
+          VELOX_USER_FAIL(
+              "Unexpected char for converting hex digit: {}!",
```

Comment:
Does the error message align with Spark?

Reply:
```
spark-sql (default)> select url_decode("AaBbC%5");
[CANNOT_DECODE_URL] Cannot decode url : AaBbC%5.
org.apache.spark.SparkIllegalArgumentException: [CANNOT_DECODE_URL] Cannot decode url : AaBbC%5.

```

Reply:
Just aligned.